spec/core_ext/hash_with_indifferent_access_spec.rb

```diff
@@ -19,6 +19,14 @@
     expect(@hash.fetch("foo", nil)).to eq("bar")
     expect(@hash.fetch(:foo)).to eq("bar")
     expect(@hash.fetch(:foo, nil)).to eq("bar")
+
+    expect(@hash.fetch("baz")).to eq("bee")
+    expect(@hash.fetch("baz", nil)).to eq("bee")
+    expect(@hash.fetch(:baz)).to eq("bee")
+    expect(@hash.fetch(:baz, nil)).to eq("bee")
+
+    expect { @hash.fetch(:missing) }.to raise_error(KeyError)
```

Comment:
@segiddins Perhaps you could change the expectation to be about an `IndexError` instead of a `KeyError`?